test/integration/kubernetes.test.ts

```diff
@@ -226,6 +244,59 @@ tap.test('snyk-monitor pulls images from a private ECR and sends data to kuberne
     'snyk-monitor sent expected data to upstream in the expected timeframe');
 });
 
+tap.test('snyk-monitor pulls images from a local registry and sends data to kubernetes-upstream', async (t) => {
+  t.teardown(async () => {
+    console.log('Begin removing local container registry...');
+    await setup.removeLocalContainerRegistry();
+    console.log('Removed local container registry');
+    console.log('Begin removing "kind" network...');
+    await setup.removeUnusedKindNetwork();
+    console.log('Removed "kind" network');
+  });
+  
+  if (process.env['TEST_PLATFORM'] !== 'kind') {
+    t.pass('Not testing local container registry because we\'re not running in KinD');
+    return;
+  }
+  t.plan(4);
+
+  const deploymentName = 'python-local';
+  const namespace = 'services';
+  const clusterName = 'Default cluster';
+  const deploymentType = WorkloadKind.Deployment;
+  const imageName = 'kind-registry:5000/python:rc-buster';
+
+  await kubectl.waitForJob('push-to-local-registry', 'default');
+  
+  console.log('Applying local registry workload...');
+  await kubectl.applyK8sYaml('./test/fixtures/insecure-registries/python-local-deployment.yaml');
+
+  console.log(`Begin polling upstream for the expected kind-registry:5000 image with integration ID ${integrationId}...`);
+
+  const validatorFn: WorkloadLocatorValidator = (workloads) => {
+    return workloads !== undefined &&
+      workloads.find((workload) => workload.name === deploymentName &&
+        workload.type === WorkloadKind.Deployment) !== undefined;
```

Comment:
Should we use the `deploymentType` variable instead of `WorkloadKind.Deployment` here?